### PR TITLE
Agnostic cloud gpu docker image and Jupyter lab

### PR DESCRIPTION
**Pull request by winglian**
This PR:

- renames the axolotl-runpod image name to axolotl-cloud to be more agnostic. We will continue in the short term to also push the same image to axolotl-runpod
- adds support for Jupyter Lab to start automatically in the cloud image. It looks for the JUPYTER_PASSWORD env var that runpod automatically sets on all containers and uses that as the default token.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ Features:
 - [Installation](#installation)
   - [Docker](#docker)
   - [Conda/Pip venv](#condapip-venv)
-  - [Runpod](#runpod)
+  - [Cloud GPU](#cloud-gpu) - Runpod, Latitude
```
Comment (NanoCode012) [inline README.md:28]: Do we need to specify the providers here? If so, does Latitude need some specific instructions like runpod?

Comment (winglian) [reply README.md:28]: I was thinking it was a good example for our use case of what those are vs say Azure & AWS

**Review by NanoCode012 — APPROVED** [review]

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -128,9 +128,10 @@ jobs:
           build-args: |
             BASE_TAG=${{ github.ref_name }}-py${{ matrix.python_version }}-cu${{ matrix.cuda }}-${{ matrix.pytorch }}${{ matrix.axolotl_extras != '' && '-' || '' }}${{ matrix.axolotl_extras }}
             CUDA=${{ matrix.cuda }}
-          file: ./docker/Dockerfile-runpod
+          file: ./docker/Dockerfile-cloud
           push: ${{ github.event_name != 'pull_request' }}
           tags: |
              ${{ steps.metadata.outputs.tags }}-py${{ matrix.python_version }}-cu${{ matrix.cuda }}-${{ matrix.pytorch }}${{ matrix.axolotl_extras != '' && '-' || '' }}${{ matrix.axolotl_extras }}
+             winglian/axolotl-runpod:main-py${{ matrix.python_version }}-cu${{ matrix.cuda }}-${{ matrix.pytorch }}${{ matrix.axolotl_extras != '' && '-' || '' }}${{ matrix.axolotl_extras }}
              ${{ (matrix.is_latest) && format('{0}-latest', steps.metadata.outputs.tags) || '' }}
```
Comment (NanoCode012) [inline .github/workflows/main.yml:136]:
```suggestion
             ${{ (matrix.is_latest) && format('{0}-latest', steps.metadata.outputs.tags) || '' }}
             ${{ (matrix.is_latest) && 'winglian/axolotl-runpod:main-latest' || '' }}
```

**Comment by NanoCode012** [comment]
```yml
  build-axolotl-runpod: # to cloud
    needs: build-axolotl
    if: github.repository_owner == 'OpenAccess-AI-Collective'
```

Does this name need to be updated also?

**Comment by winglian** [comment]
Nah. That's for the Github action to run.

**Pull request merged**